provider/cloudflare/cloudflare.go

```diff
@@ -557,6 +579,17 @@ func (p *CloudFlareProvider) AdjustEndpoints(endpoints []*endpoint.Endpoint) ([]
 		}
 		e.SetProviderSpecificProperty(source.CloudflareProxiedKey, strconv.FormatBool(proxied))
 
+		if p.CustomHostnamesConfig.Enabled {
+			// sort custom hostnames in annotation to properly detect changes
+			if customHostnames := getEndpointCustomHostnames(e); len(customHostnames) > 1 {
+				sort.Strings(customHostnames)
```

Comment:
why not to sort in `getEndpointCustomHostnames` method itself?

Reply:
actually we only need to sort it here, just removed it from `getEndpointCustomHostnames()` to avoid redundant sorting for each function call (see comment below for more details)